Source/WebCore/platform/graphics/ComplexTextController.cpp

```diff
@@ -314,7 +314,21 @@ void ComplexTextController::collectComplexTextRuns()
         return characters;
     }();
 
+    auto isJapanese = [](const String& str) {
+        StringView textView = str;
+        for (size_t i = 0; i < textView.length(); ++i) {
+            auto ch = textView[i];
+            if ((ch >= 0x4E00 && ch <= 0x9FFF) || (ch >= 0x30A0 && ch <= 0x30FF) || (ch >= 0x4E00 && ch <= 0x9FFF))
+                return true;
+        }
+        return false;
```

Comment:
I think it is better to have such functions in a place like `Source/WTF/wtf/text/CharacterProperties.h` or `TextUtil.h`. This is specially important because you need to cover both our code paths: `ComplexTextController` and `WidthIterator` since we handle small-caps on both of them, unless we can guarantee that we will always pick the same path for such content.